### PR TITLE
Fix Grid-RCNN error when there is no bbox result

### PR DESCRIPTION
**Pull request by jiangwenj02**
## Motivation
bbox_results.append(grid_rois.new_tensor([])) will cause problem when we eval the grid rcnn and it det_bboxes[i].shape[0] == 0.
the type of  bbox_results[*] should be numpy.

## Modification
we modify it to numpy.

### CODE REVIEW

File: mmdet/models/roi_heads/grid_roi_head.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def simple_test(self,
             num_imgs = len(det_bboxes)
             for i in range(num_imgs):
                 if det_bboxes[i].shape[0] == 0:
-                    bbox_results.append(grid_rois.new_tensor([]))
+                    bbox_results.append([np.zeros((0, 5), dtype=np.float32) for i in range(self.bbox_head.num_classes)])
```
Comment (ZwwWayne) [inline mmdet/models/roi_heads/grid_roi_head.py:157]: Why it is numpy array rather than tensor? can we use `grid_rois_new_tensor(size=(0,5))`?

**Review by hhaAndroid — APPROVED** [review]

**Comment by hhaAndroid** [comment]
Hi @jiangwenj02 Thank you for your contribution.  Can you sign the CLA, and fix the lint format problem.

**Comment by jiangwenj02** [comment]
Hi,  I am not very familiar with pull request, and i will try solve it.

Haian Huang(深度眸) ***@***.***> 于2021年6月17日周四 下午1:17写道：

**Comment by jiangwenj02** [comment]
Because the situation is rarely, the modified line is not covered by test file. How should i do?

**Comment by ZwwWayne** [comment]
Hi @jiangwenj02 ,
Thanks, it should be fine and this PR will be merged.

**Review by ZwwWayne — APPROVED** [review]

**Comment by OpenMMLab-Assistant003** [comment]
Hi @jiangwenj02！First of all, we want to express our gratitude for your significant PR in the MMDetection project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤ @jiangwenj02

**Comment by jiangwenj02** [comment]
I accept the invitation.

OpenMMLab-Assistant 003 ***@***.***> 于 2023年4月7日周五 上午10:48写道：

**Pull request merged**